docs/source/torchao_hf_integration.md

```diff
@@ -0,0 +1,380 @@
+| Integer X-bit quantization | `IntxWeightOnlyConfig` |
+| Floating point X-bit quantization   | `FPXWeightOnlyConfig` |
+| Unsigned Integer Quanization   | `GemliteUIntXWeightOnlyConfig` <br> `UIntXWeightOnlyConfig`|
+
+```{note}
+For full definitions of the above types, please see [here](https://github.com/pytorch/ao/blob/main/torchao/quantization/quant_api.py)
+```
+
+
+(Serialization)=
+## Serialization
+
+To serialize a quantized model in a given dtype, first load the model with the desired quantization dtype and then save it using the `save_pretrained()` method.
+
+
+**Using Transformers**:
+```python
+import torch
+from transformers import TorchAoConfig, AutoModelForCausalLM, AutoTokenizer
+from torchao.quantization import Int8WeightOnlyConfig
+
+quant_config = Int8WeightOnlyConfig(group_size=128)
+quantization_config = TorchAoConfig(quant_type=quant_config)
+
+# Load and quantize the model
+quantized_model = AutoModelForCausalLM.from_pretrained(
+    "meta-llama/Llama-3.1-8B-Instruct",
+    dtype="auto",
+    device_map="cpu",
+    quantization_config=quantization_config
+)
+# save the quantized model
+output_dir = "llama-3.1-8b-torchao-int8"
+quantized_model.save_pretrained(output_dir, safe_serialization=False)
+```
+
+**Using Diffusers**:
+```python
+import torch
+from diffusers import AutoModel, TorchAoConfig
+
+quantization_config = TorchAoConfig("int8wo")
+transformer = AutoModel.from_pretrained(
+    "black-forest-labs/Flux.1-Dev",
+    subfolder="transformer",
+    quantization_config=quantization_config,
+    torch_dtype=torch.bfloat16,
+)
+transformer.save_pretrained("/path/to/flux_int8wo", safe_serialization=False)
+```
+
+
+To load a serialized quantized model, use the `from_pretrained()` method.
+
+**Using Transformers**:
+```python
+# reload the quantized model
+reloaded_model = AutoModelForCausalLM.from_pretrained(
+    output_dir,
+    device_map="auto",
+    dtype=torch.bfloat16
+)
+tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-3.1-8B-Instruct")
+input_text = "What are we having for dinner?"
+input_ids = tokenizer(input_text, return_tensors="pt").to(reloaded_model.device.type)
+
+output = reloaded_model.generate(**input_ids, max_new_tokens=10)
+print(tokenizer.decode(output[0], skip_special_tokens=True))
+```
+
+**Using Diffusers**:
+```python
+import torch
+from diffusers import FluxPipeline, AutoModel
+
+transformer = AutoModel.from_pretrained("/path/to/flux_int8wo", torch_dtype=torch.bfloat16, use_safetensors=False)
+pipe = FluxPipeline.from_pretrained("black-forest-labs/Flux.1-Dev", transformer=transformer, torch_dtype=torch.bfloat16)
+pipe.to("cuda")
+
+prompt = "A cat holding a sign that says hello world"
+image = pipe(prompt, num_inference_steps=30, guidance_scale=7.0).images[0]
+image.save("output.png")
```

Comment:
Looks like this is duplicated again, maybe we don't need a serialization section